### DISCUSSION TITLE
How to specify BLOCKSIZE in `to_raster()`?

### DISCUSSION THREAD

**Discussion by rsignell-usgs** [corteva/rioxarray#656] [Q&A] [answered]
I have an xarray dataarray, and I can write to COG using rioxarray using the default 512x512, but then I tried to specify the blocks to be 1024 instead of 512.   I thought this would just be:
``` 
da.rio.to_raster(raster_path=cog, driver="COG", profile_kwargs={'BLOCKSIZE':1024})
```
but it just seems to ignore it and generates 512x512 again, so what did I do wrong? 

The full notebook is here (which is supposed to demonstrate how to work with big geotiff using Dask):
https://nbviewer.org/gist/rsignell-usgs/e60a6890ef7f5e0e890acc3af0f8da4e

**Comment by snowman2** [comment, marked as answer]
https://corteva.github.io/rioxarray/stable/rioxarray.html#rioxarray.raster_array.RasterArray.to_raster

`profile_kwargs` are keyword arguments. So, you would input them like so:

```python
da.rio.to_raster(raster_path=cog, driver="COG", blocksize=1024)
```

**Reply by rsignell-usgs** [reply]
Here's the updated notebook!  Please let me know if you see any improvements that could be made!
https://nbviewer.org/gist/rsignell-usgs/aa0b4d6f38a0c84a9c11deab3fac8a2a
